src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -549,6 +566,12 @@ private void WriteStatusCodeSwitch(CodeWriter writer, CodeWriterDeclaration mess
                             writer.Line($"var {valueVariable:D} = {message}.ExtractResponseContent();");
                             value = new Reference(valueVariable.ActualName, responseBody.Type);
                         }
+                        else if (responseBody is ConstantResponseBody _ && returnType != null)
```

Comment:
Don't discard the ConstantResponseBody variable here and avoid the cast on the next line.